### PR TITLE
MNT: Unpin stuff, update matrix

### LINKED ISSUE #6: MNT: Unpin tox < 4 when we can
I pinned it to `tox < 4` in `tox.ini` because I don't have time to deal with breaking changes right now.

### LINKED ISSUE #7: Unpin SQLAlchemy<2 when sunpy lets us
https://github.com/astropy/astropy-integration-testing/blob/928e89f5fd7145a7249d7d84914f3e898d8862f1/tox.ini#L37

Blocked by:

* https://github.com/sunpy/sunpy/issues/6647

### PR DESCRIPTION
**Pull request by pllim**
Might as well see where we're at while testing both numpy and matplotlib RCs.

Fix #6 

Fix #7

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -34,8 +33,7 @@ deps =
     reproject,all: git+https://github.com/astropy/reproject.git#egg=reproject[test,all]
     specreduce,all: git+https://github.com/astropy/specreduce.git#egg=specreduce[test]
     specutils,all: git+https://github.com/astropy/specutils.git#egg=specutils[all,test]
-    sunpy,all: SQLAlchemy<2
-    sunpy,all: git+https://github.com/sunpy/sunpy.git#egg=sunpy[tests,all]
+    sunpy,all: git+https://github.com/sunpy/sunpy.git#egg=sunpy[tests,asdf,dask,database,image,jpeg2000,map,net,timeseries,visualization]
```
Comment (astrofrog) [inline tox.ini:36]: Does all not work? (Cc @Cadair)

**Comment by pllim** [comment]
I am also not sure what to do with the astroquery failures. They are all about not catching deprecation warnings. Does not seem in scope to fix here, though ✔️ would be nice.

**Comment by bsipocz** [comment]
We do catch deprecation warnings, and devtest ci was passing earlier this week. Open one upstream if it's real, I maybe able to come back to it early next week.

**Comment by pllim** [comment]
Yeah, I don't think this is astroquery bug. I am just not sure how to set up the pytest settings here to be compatible with what astroquery is doing. Ideas welcome.

**Pull request closed without merging**